### PR TITLE
Add new platform versions for WWDC24

### PR DESCRIPTION
**Pull request by liamcharger**
This PR adds new iOS, tvOS, macOS, and visionOS platform versions to support the new versions of the said OSes, announced at WWDC24.

Closes: #419

### CODE REVIEW

File: Sources/ViewTypes/Window.swift [hunk 1]
```diff
@@ -78,7 +79,7 @@ extension tvOSViewVersion<WindowType, UIWindow> {
     public static let v14 = Self(for: .v14, selector: selector)
     public static let v15 = Self(for: .v15, selector: selector)
     public static let v16 = Self(for: .v16, selector: selector)
-    public static let v17 = Self(for: .v17, selector: selector)
```
Comment (shashwatkn) [inline Sources/ViewTypes/Window.swift:81]: Is this expected?

File: Sources/ViewTypes/TextFieldWithVerticalAxis.swift [hunk 2]
```diff
@@ -96,11 +97,12 @@ extension tvOSViewVersion<TextFieldWithVerticalAxisType, UITextField> {
     public static let v15 = Self.unavailable()
 
     public static let v16 = Self(for: .v16)
-    public static let v17 = Self(for: .v17)
```
Comment (shashwatkn) [inline Sources/ViewTypes/TextFieldWithVerticalAxis.swift:99]: Is this expected?

File: Sources/ViewTypes/Window.swift (hunk 1 above)
Comment (shashwatkn) [inline Sources/ViewTypes/Window.swift:81]: +1

**Comment by sarthak-mishra-swiggy** [comment]
@davdroman Can we get this merged? Its breaking for some of our users who have upgraded to the first beta of iOS 18. I am sure a lot of other devs would be facing this as well. Thanks!

File: Sources/ViewTypes/TextFieldWithVerticalAxis.swift (hunk 2 above)
Comment (liamcharger) [reply Sources/ViewTypes/TextFieldWithVerticalAxis.swift:99]: Oops, that was a mistake, thanks for catching it!

File: Sources/ViewTypes/DatePickerWithCompactStyle.swift [hunk 3]
```diff
@@ -73,10 +73,12 @@ extension iOSViewVersion<DatePickerWithCompactStyleType, UIDatePicker> {
     public static let v15 = Self(for: .v15)
     public static let v16 = Self(for: .v16)
     public static let v17 = Self(for: .v17)
+    public static let v18 = Self(for: .v18)
 }
 
 extension visionOSViewVersion<DatePickerWithCompactStyleType, UIDatePicker> {
     public static let v1 = Self(for: .v1)
+    public static let v1 = Self(for: .v2)
```
Comment (KKirsten) [inline Sources/ViewTypes/DatePickerWithCompactStyle.swift:81]:
```suggestion
    public static let v2 = Self(for: .v2)
```

**Comment by aarishKhanna04** [comment]
+1 please get this merged.

**Comment by Dimillian** [comment]
But does this validate that the underlying components are still the same etc..?

**Comment by arnauddorgans** [comment]
+1

**Comment by frlefebvre** [comment]
Yes please 🙏

**Pull request merged**